### ISSUE TITLE
How to ignore copyrights in HTML from compressing?

### ISSUE THREAD

**Issue by julkue** [searchturbine/phpwee-php-minifier#1]
Hello,

my copyrights look like 

``` html
<!--
 /**
  * Some copyright
  *
  */
-->
```

How can I ignore those from compressing HTML?

**Comment by ztkemsley** [comment]
The short answer is putting [] square brackets inside the comment will keep it included - and its whitespace intact.

```
<!--[
 /**
  * Some copyright
  *
  */
]-->
```

The alternative - we can edit HTMLmin.php to keep the comment:

Line 50:

```
    foreach ($xpath->query('//comment()') as $comment) {
                $val= $comment->nodeValue;
                if( strpos($val,'[')!==0 &&  strpos(strtolower($val),'copyright')!==0 ){
                    $comment->parentNode->removeChild($comment);
                }
            }
```

**Comment by julkue** [comment]
Thanks for the hint. But isn't it possible to pass a individual structure for copyrights as a parameter?

**Comment by ztkemsley** [comment]
Its a good idea.  I can add that feature no problem.

For now you could just so a  str_replace

```
 str_replace("</body>",$mynotice.”</body>”,$html)
```

**Comment by julkue** [comment]
Yes but my problem is that the copyrights are in the HTML itself, so I would replace them with a regex first and store them to a temporary variable. After the minify I would need to add them again. Therefore I would need to know where to put them. A bit complex I think.

**Comment by ztkemsley** [comment]
Sure - so the logic you want is to keep all of your html comments that contain the word copyright?
Perhaps a parameter set regex matches for comments to keep?

We already do have a system for identifying comments you want to keep:

```
<!--[
 /**
  * Some copyright
  *
  */
]-->
```

Can you explain why that is unworkable so i can understand?
